src/de/schildbach/pte/MvvProvider.java

```diff
@@ -157,4 +166,117 @@ protected Position parsePosition(final String position) {
+            context.from = from;
+            context.to = to;
+            context.via = via;
+            context.options = options;
+            context.trips = result.trips;
+            result = new QueryTripsResult(result.header, result.queryUri, result.from,
+                    result.via, result.to, context, result.trips);
+        }
+        return result;
+    }
+
+    @Override
+    public QueryTripsResult queryMoreTrips(final QueryTripsContext contextObj, final boolean later) throws IOException {
+        if (!(contextObj instanceof MvvContext)) {
+            throw new IllegalArgumentException("needs an MvvContext");
+        }
+        MvvContext context = (MvvContext) contextObj;
+
+        // get departure time of last trip / arrival time of last trip as reference time
+        int tripIndex;
+        if (later) {
+            Trip lastTrip = context.trips.get(context.trips.size() - 1);
+            // if the last included trip is a walking route, use the previous one
+            boolean lastTripIsIndividual =
+                    lastTrip.legs.size() == 1 && lastTrip.legs.get(0) instanceof Trip.Individual;
+            tripIndex = context.trips.size() - (lastTripIsIndividual ? 2 : 1);
+        } else {
+            tripIndex = 0;
+        }
+        Trip refTrip = context.trips.get(tripIndex);
+        Date refTime = later ? refTrip.getFirstDepartureTime() : refTrip.getLastArrivalTime();
+
+        // adjust time by one minute so that we don't get the same trip again
+        refTime = addMinutesToDate(refTime, later ? 1 : -1);
+
+        requestingMoreTrips = true; // set special options for more trips request
+        try {
+            QueryTripsResult result = super.queryTrips(context.from, context.via, context.to,
+                    refTime, later, context.options);
+
+            if (result.status == QueryTripsResult.Status.OK) {
+                context.trips.addAll(later ? context.trips.size() - 1 : 0, result.trips);
+                result = new QueryTripsResult(result.header, result.queryUri, result.from,
+                        result.via, result.to, context, result.trips);
+            }
+
+            return result;
+        } finally {
+            requestingMoreTrips = false;  // reset options
+        }
+    }
+
+    @Override
+    protected void appendTripRequestParameters(HttpUrl.Builder url, Location from, @Nullable Location via, Location to, Date time, boolean dep, @Nullable TripOptions options) {
+        super.appendTripRequestParameters(url, from, via, to, time, dep, options);
+
+        if (requestingMoreTrips) {
+            // ensure that the first displayed trip is after the given departure time /
+            // last displayed trip is before the given arrival time
+            url.addEncodedQueryParameter("calcOneDirection", "1");
+        }
+    }
+
+    private Date addMinutesToDate(Date initial, int minutes) {
+        Calendar c = Calendar.getInstance();
```

Comment:
Maybe  `Calendar c = new GregorianCalendar(timeZone);`? The time zone for MvvProvider is CET.

Reply:
Fixed ✅